js/add_student.js

```diff
@@ -1,101 +1,207 @@
-sheetHeaders = array[0];
-objectArray = arrayToObjects(array);
-
-
-function arrayToObjects(array) {
-    let headers = array[0];
-    let result = [];
-    let tempStr = "";
-    for (let i = 1; i < array.length; i++) {
-        tempStr = "{";
-        for (let j = 0; j< array[i].length; j++) {
-            tempStr += "\"" + headers[j] + "\":\"" + array[i][j] + "\"";
-            if (j < array[i].length - 1) {
-                tempStr += ",";
+
+function dispHeaders() {
+    let textField;
+    for(let i = 0; i < uplsHeaders.length; i++){
+        let headerDiv = document.createElement("div");
+        headerDiv.className = "ex1";
+        headerDiv.setAttribute("id", "div"+i);
+        let txt = document.createTextNode(uplsHeaders[i]);
+        if (uplsHeaders[i] == "FROZEN_UPLS_ACCOUNT") { continue;}
+        headerDiv.appendChild(txt);
+
+        if(uplsHeaders[i] == "LIBRARIAN"){
+            textField = document.createElement("div");
+            let insertLib = document.createTextNode(lib);
+            textField.appendChild(insertLib);
+        }
+        else if(uplsHeaders[i] == "PROGRAM_DESCRIPTION") {
+            textField = document.createElement("select");
+            textField.style.width = "250px";
+            for(let i = 1; i < progArray.length; i++) {
+                if(progArray[i] == undefined) {continue;}
+                let option = document.createElement("option");
+                option.value = progArray[i];
+                option.text = progArray[i];
+                textField.appendChild(option);
+            }
+        }
+        else if(uplsHeaders[i] == "GENERAL_COMMENT") {
+            textField = document.createElement("textarea");
+        }
+        else if(uplsHeaders[i] == "CLASS_LEVEL") {
+            textField = document.createElement("input");
+            textField.type = "number";
+        }
+        else if(uplsHeaders[i] == "DATE_ADDED_TO_SYSTEM"){
+            textField = document.createElement("div");
+            let insertDate = document.createTextNode(currentDate);
+            textField.appendChild(insertDate);
+
+        }
+        else if(uplsHeaders[i] == "STUDENT_ID"){
+            textField = document.createElement("input");
+            textField.type = "number";
+        }
+        else if(uplsHeaders[i]  == "EMAIL"){
+            textField = document.createElement("input");
+            textField.type = "email";
+        }
+        else if(i>15 && typeArray[i] == "CheckBox"){
+            textField = document.createElement("input");
+            textField.type = "checkbox";
+        }
+        else if(i>15 && typeArray[i] == "Date"){
+            textField = document.createElement("input");
+            textField.type = "date";
+        }
+        else if(i>15 && typeArray[i] == "Number"){
+            textField = document.createElement("input");
+            textField.type = "number";
+        }
+        else if(i>15 && typeArray[i] == "Text" && vocabsArray[i] !== undefined){
```

Comment:
no need to check the type if there is a vocabulary...any field with limited vocabulary will be a select box